documentation/messages/user.password_reset.transactional.md

```diff
@@ -0,0 +1,109 @@
+# User password reset transactional message
+
+User password reset event to trigger functionality within the Quicksilver system.
+
+----
+
+#### Current Configuration
+Accessed by connecting to RabbitMQ server and sending message in the following format.
+
+**Routing key**
+
+`user.password_reset.transactional`
+
+**Exchanges**
+- `transactionalExchange`
+
+**Consumers**
+
+| Consumer                                                                                 | Binding key           |
+| ---------------------------------------------------------------------------------------- | ----------------------|
+| [mbp-externalApplications-dashboard](../consumers/mbp-externalApplications-dashboard.md) |  `*.*.transactional`  |
+| [mbc-logging-gateway](../consumers/mbc-logging-gateway.md)                               |  `*.*.transactional`  |
+| [mbc-transactional-email](../consumers/mbc-transactional-email.md)                       |  `*.*.transactional`  |
+
+#### Current Message Payload
+
+```js
+{
+
+  /* Required. Always must be `user_password`. */
+  activity: String,
+
+  /* Required. */
+  email: String,
+
+  /* Required. Phoenix user id. */
+  uid: Number,
+
+  /* Required, Example: `US`. */
+  user_country: String,
+
+  /* Required, Example: `en`. */
+  user_language: String,
+
+  /* Optional. Default is generated base on `user_country` field. Example: `mb-user-password-US`. */
+  email_template: String,
+
+  /* Required. Unix timestamp. Example: `1458067430`. */
+  activity_timestamp: Number,
+
+  /* Required. Example: `US`. */
+  application_id: String,
+
+  /* Required. An array of string to tag the message with. Example: `drupal_user_password`. */
+  email_tags: Array,
+
+  /* Required. An array of variables to inject into Mandrill template. */
+  /* Required keys: `MEMBER_COUNT`, `FNAME`, 'RESET_LINK' */
+  merge_vars: Array,
+
+}
+```
+
+----
+
+#### Suggested API
+
+**Endpoint**
+
+`POST /user/password`
+
+**Body Parameters**
+
+```js
+{
+
+  /* Required: user_id or email or mobile. Northstar user ID. Example: `555b9225bffebc31068b4567`. */
+  user_id: String,
+
+  /* Required: user_id or email or mobile. */
+  email: String,
+
+  /* Required: user_id or email or mobile. */
+  mobile: Integer,
+
+  /* Required. Valid reset password url. Example: `https://www.dosomething.org/user/reset/1/1458839724/N1RILd4CdbNExmQqcDQWi13KYTI-gQ1URWKmwCvFAfE`. */
```

Comment:
The resetURL is generated by the consumer that responds to this message. This is done as there are message requests from outside of the Drupal app that need this functionality: `mbc-user-import`.

- https://github.com/DoSomething/mbc-user-import/blob/master/src/MBC_UserImport_Toolbox.php#L239
- https://github.com/DoSomething/mb-toolbox/blob/master/src/MB_Toolbox.php#L251-L276

Perhaps this should be an optional?